framework/Rdo/lib/Horde/Rdo/Base.php

```diff
@@ -446,4 +446,30 @@ protected function _fillPlaceholders($query)
         return $query;
     }
 
+    /**
+     * make the Entity convertible to an Array
+     * this method can be used when handing it over to Horde_Variables
+     * so that the database is not unnecessarily queried because of lazyFields/-Relationships
+     *
+     * @param bool $lazy            if lazy elements should be added
+     * @param bool $relationships   if relationships should be added
+     *
+     * @return array containing all selected fields / relationships
+     */
+    public function toArray($lazy = false, $relationships = false)
+    {
+        $array = array();
+
+        $m = $this->getMapper();
+        $keys = array_merge($m->fields,
+                            $lazy          ?            $m->lazyFields         : array(),
+                            $relationships ? array_keys($m->relationships)     : array(),
+                   $lazy && $relationships ? array_keys($m->lazyRelationships) : array());
+
+        foreach ($keys as $field) {
+            $array[$field] = $this->$field;
+        }
+
+        return $array;
+    }
 }
```

Comment:
This could be simplified and made more readable. Since you loop through the keys in the end anyway, do this right from the start, each of the field lists in a separate if-clause. I.e. `foreach ($m->fields as $field)... if ($lazy) { foreach ($m->lazyFields as $field)` You get the idea.